src/prommis/HD_CE.py

```diff
@@ -0,0 +1,344 @@
+        Q3_spec = self.sample_cp*(self.operating_temperature - 298.15)
+        Q3_R =  (Q3/(self.ramp_up_time*60))                                                                                     #KJ/KG of sample
+
+        Actual_coil_rating = (Q1_R+Q3_R)/self.efficiency
+        M = [Q1_spec, Q2_spec, Q3_spec, Actual_coil_rating]
+
+        return  M
+    
+    def HDF_power(self):
+        
+        Q2 = self.heat_loss()[5]                                                                                                                            #Heat loss from the external surface area of a furnace
+        power = Q2/self.efficiency
+        return power
+    
+    def HDF_OPEX(self):
+        M = self.energy_req()
+
+        Q_total = M[0] + M[1] + M[2]                                                                                                                        #KJ/KG of sample
+        total_runs = (self.hours_per_shift*self.shifts_per_day*self.operating_days_per_year)/self.decrepitation_duration                                    #Total number of runs in a year 
+        total_energy_required = (Q_total * total_runs)/3600
+        utility = (total_energy_required * self.electricity_rate)                                                                                           #USD Per kg of sample
+        processing_time = (self.preparation_time + (self.ramp_up_time/60) + self.decrepitation_duration+self.cool_down_time)/self.sample_mass               # hr/kg
+        opex = (1/processing_time) * (utility) * (8000)                                                                                                     #USD per year
+
+        return opex                                                              
+        
+    def heating_coil(self):
+        M = self.energy_req()[3]
+        if 0<M <=5:
+            price = 80
+        elif M>5:                                                                
+            price = (14.81*M) + 5.93
+        return price
+    
+    def HDF_CAPEX(self):
+
+        """
+        transformer_cost specification - Transformer, dry type, DOE 2016, 30kVA, 3 phase, 480V delta primary, 208Y/120V secondary, 18M, 150C rise
+        temperature_pressure_controller_price - N480D PID Temperature controller
+        """
+
+        vessel_weight = self.C_Ins1()[0]+self.C_Metal1()[0]+self.C_Ins2()[0]+self.C_Metal2()[0]
+        vessel_volume = self.C_Metal2()[4]*35.315
+        vessel_density = vessel_weight/vessel_volume
+        transportation_cost = (10+(0.57*vessel_density))*self.labor_rate 
+        inspection_cleaning_and_testing = (50+(1.892*self.C_Metal2()[3]))*self.labor_rate
+
+        CAPEX = self.C_Ins1()[1] + self.C_Metal1()[1] + self.C_Ins2()[1]+ self.C_Metal2()[1] + self.transformer_cost + self.temperature_controller_price+self.heating_coil() + transportation_cost + inspection_cleaning_and_testing                                             
```

Comment:
I am somewhat uncertain as to what the purpose of this file is - for the most part this is just doing Python operations which does not fit in with the rest of the code which works builds Pyomo models. Could you clarify the purpose of this code a bit more?